scipy/special/ellint_carlson/ellint_compat.h

```diff
@@ -0,0 +1,180 @@
+#ifndef _ELLINT_COMPAT_H_INCLUDED
```

Comment:
TBH, I'm not a big fan about this --- it would seem quite better to write this in c++ insofar as std::complex goes, instead of emulating complex numbers with macros.

Reply:
> TBH, I'm not a big fan about this --- it would seem quite better to write this in c++ insofar as std::complex goes, instead of emulating complex numbers with macros.

This can be done, but it will take a bit of time, as I dust my memories of the language.